File: LCA.js
class Entity {}

class Object extends Entity {}

class Property extends Entity {}

class Relation extends Entity {
  constructor(subject, object) {
    this.subject = subject
    this.object = object
  }
}

class Correspondence extends Entity {
  constructor(property1, property2) {
    this.property1 = property1
    this.property2 = property2
  }
}

class LinearCorrespondence extends Correspondence {
  isLinearCorrespondence(relation1, relation2) {
    for (let i = 0; i < relation1.subject; i++) {
      for (let j = 0; j < relation1.object; j++) {
        for (let k = 0; k < relation2.object; k++) {
          if (relation1[i][j] && relation2[j][k]) {
            if (
              !(
                this.property1[i][j] == this.property1[j][k] &&
                this.property2[i][j] == this.property2[j][k]
              )
            ) {
              return false
            }
          }
        }
      }
    }
    return true
  }
}

const john = new Object()
const mary = new Object()
const book = new Object()
const table = new Object()

const owns = new Property()
constreads = new Property()

const r1 = new Relation(john, book)
const r2 = new Relation(book, table)

const c = new Correspondence(owns, reads)
const lc = new LinearCorrespondence(owns, reads)

if (lc.isLinearCorrespondence(r1, r2)) {
  console.log('The linear correspondence axiom holds!')
} else {
  console.log('The linear correspondence axiom does not hold.')
}

File: LCA.cs
using System;

class Entity { }

class Object : Entity { }

class Property : Entity { }

class Relation : Entity
{
    public Object Subject { get; set; }
    public Object Object { get; set; }
}

class Correspondence : Entity
{
    public Property Property1 { get; set; }
    public Property Property2 { get; set; }
}

class LinearCorrespondence : Correspondence
{
    public bool IsLinearCorrespondence(Relation relation1, Relation relation2)
    {
        for (int i = 0; i < relation1.Subject; i++)
        {
            for (int j = 0; j < relation1.Object; j++)
            {
                for (int k = 0; k < relation2.Object; k++)
                {
                    if (relation1[i, j] && relation2[j, k])
                    {
                        if (!(Property1[i, j] == Property1[j, k] && Property2[i, j] == Property2[j, k]))
                        {
                            return false;
                        }
                    }
                }
            }
        }
        return true;
    }
}

class Program
{
    static void Main(string[] args)
    {
        Object john = new Object();
        Object mary = new Object();
       Object book = new Object();
        Object table = new Object();

        Property owns = new Property();
        Property reads = new Property();

        Relation r1 = new Relation { Subject = john, Object = book };
        Relation r2 = new Relation { Subject = book, Object = table };

        Correspondence c = new Correspondence { Property1 = owns, Property2 = reads };

        LinearCorrespondence lc = new LinearCorrespondence { Property1 = owns, Property2 = reads };

        if (lc.IsLinearCorrespondence(r1, r2))
        {
            Console.WriteLine("The linear correspondence axiom holds!");
        }
        else
        {
            Console.WriteLine("The linear correspondence axiom does not hold.");
        }
    }
}